src/client/components/pages/parts/privs-edit-modal.js

```diff
@@ -0,0 +1,153 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import {PrivilegeTypes, getPrivilegeShieldIcon, getPrivilegeTitleFromBit} from '../../../../common/helpers/privileges-utils';
+import {faPencilAlt, faTimes} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PrivilegeBadges from './privilege-badges';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+
+const {Button, Form, Modal} = bootstrap;
+
+class PrivsEditModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			privs: props.targetUser.privs,
+			submittable: false
+		};
+
+		this.handleBitChange = this.handleBitChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+	async handleSubmit() {
+		const {privs} = this.state;
+		const oldPrivs = this.props.targetUser.privs;
+		if (privs === oldPrivs) {
+			return;
+		}
+
+		const data = {
+			adminId: this.props.adminId,
+			newPrivs: privs,
+			oldPrivs,
+			targetUserId: this.props.targetUser.id
+		};
+
+		try {
+			const response = await fetch('/editor/privs/edit/handler', {
+				body: JSON.stringify(data),
+				headers: {
+					'Content-Type': 'application/json; charset=utf-8'
+				},
+				method: 'POST'
+			});
+
+			if (!response.ok) {
+				const {error} = await response.json();
+				throw new Error(error ?? response.statusText);
+			}
+			this.props.updateResultsOnPrivsChange();
+			this.props.handleCloseModal();
+		}
+		catch (err) {
+			throw new Error(err);
+		}
+	}
+
+	/* eslint-disable no-bitwise */
+	handleBitChange(bit) {
+		const newPrivs = this.state.privs ^ (1 << bit);
+		if (this.props.targetUser.privs !== newPrivs) {
+			this.setState({
+				privs: newPrivs,
+				submittable: true
+			});
+		}
+		else {
+			this.setState({
+				privs: newPrivs,
+				submittable: false
+			});
+		}
+	}
+
+	/* eslint-disable react/jsx-no-bind */
+	render() {
+		const link = `/editor/${this.props.targetUser.bbid}`;
+
+		const switches = Object.values(PrivilegeTypes).map(priv => (
+			<Form.Check
+				checked={this.state.privs & priv.value}
+				id={`bit${priv.bit}`}
+				key={priv.bit}
+				label={getPrivilegeTitleFromBit(priv.bit)}
+				type="switch"
+				onChange={() => this.handleBitChange(priv.bit)}
+			/>
+		));
+
+		return (
+			<Modal
+				scrollable
+				show={this.props.show}
+				onHide={this.props.handleCloseModal}
+			>
+				<Modal.Header closeButton>
+					<Modal.Title>
+						<img height="20" src={getPrivilegeShieldIcon(this.state.privs)}/> {' '}
+						<a href={link}>
```

Comment:
I wonder if this should be a link to the  user's profile. It might be clearer to have this be just text, and have a separate "visit profile" link in the modal body (perhaps opening in a new tab).
What do you think?

Reply:
For now, I have removed the link to the user's profile in this follow up [PR](https://github.com/metabrainz/bookbrainz-site/pull/998). I will, however, be sure to take care of this "Visit Profile" functionality in the next iteration of the PrivsModal.